### PR TITLE
fixed accessibility issue related to rangeinput and carousel

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: gurkiran_singh <gurkiransinghk@gmail.com>

#### What does this PR do?
Accessibility fixes for `Carousel` and `RangeInput`.

#### Where should the reviewer start?
componentts => Carousel.js, StyledRangeInput.js

#### What testing has been done on this PR?
yarn test-update

#### How should this be manually tested?
Storybook

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
#4068 
Fixes : -
### Carousel
- [x] - Screen reader users should clearly know which slide is being shown (currently, tabbing to the control buttons just says "Subtract button")
### RangeInput
- [x] - Focus should be placed on the rangeinput thumb (visual input the mouse would move) 

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
N/A

#### Is this change backwards compatible or is it a breaking change?
N/A

### CODE REVIEW

File: src/js/components/RangeInput/StyledRangeInput.js [hunk 1]
```diff
@@ -108,6 +108,7 @@ const firefoxMicrosoftThumbStyle = css`
   ${rangeThumbStyle} margin-top: 0px;
   height: ${props => props.theme.global.spacing};
   width: ${props => props.theme.global.spacing};
+  ${props => props.focus && focusStyle()}
```
Comment (taysea) [inline src/js/components/RangeInput/StyledRangeInput.js:111]: I am not seeing the focus indicator on the thumb. Instead, I just see the browser default around the entire range input. it seems like the focus is still on the overall element. I am on MacOS Chrome:
Screen Shot 2021-05-14 at 9 41 54 AM

Comment (g4rry420) [reply src/js/components/RangeInput/StyledRangeInput.js:111]: Actually, I am on Windows Firefox and it's working on it but not on the Chrome. I take a look into it.

Comment (g4rry420) [reply src/js/components/RangeInput/StyledRangeInput.js:111]: Also,  shall I remove the browser default focus indicator or not ?

**Comment by g4rry420** [comment]
@taysea  I have made some changes, Please let me know about these.

**Comment by taysea** [comment]
> @taysea I have made some changes, Please let me know about these.

Looks good on Mac Firefox and Chrome -- focus indicator is showing on the thumb and no browser default shows around the entire input, but on Safari I still see the browser default on the slider:
Screen Shot 2021-05-17 at 10 27 24 AM

**Comment by g4rry420** [comment]
@taysea I made some changes and believe this should work on safari too. I haven't test them. Please let me know.

**Comment by taysea** [comment]
> @taysea I made some changes and believe this should work on safari too. I haven't test them. Please let me know.

I am still seeing the browser default on safari

**Comment by g4rry420** [comment]
@taysea I have made changes but I am unable to test it as I am on windows and safari is not supported by windows. If you still see the browser default, can you please make changes as required.

**Review by taysea — APPROVED** [review]
This looks good now!

**Comment by ericsoderberghp** [comment]
I'm seeing two focus indicators when in a FormField context here: https://deploy-preview-5254--sad-tereshkova-173d07.netlify.app/?path=/story/input-form-controlled--controlled. I suspect we might need a `focusIndicator={false}`

**Comment by taysea** [comment]
> I'm seeing two focus indicators when in a FormField context here: https://deploy-preview-5254--sad-tereshkova-173d07.netlify.app/?path=/story/input-form-controlled--controlled. I suspect we might need a `focusIndicator={false}`

Good catch, there's a double focus in the published storybook as well but might be nice to clean that up here.

**Comment by g4rry420** [comment]
> I'm seeing two focus indicators when in a FormField context here: https://deploy-preview-5254--sad-tereshkova-173d07.netlify.app/?path=/story/input-form-controlled--controlled. I suspect we might need a `focusIndicator={false}`

I tried `focusIndicator={false}` but it doesn't seem to works. Also I have noticed that double focus error also exists previously as in #4947.

File: src/js/components/Carousel/Carousel.js [hunk 2]
```diff
@@ -112,7 +112,10 @@ const Carousel = ({
         // eslint-disable-next-line react/no-array-index-key
         key={index}
         icon={
-          <CurrentIcon color={activeIndex === index ? iconColor : undefined} />
+          <CurrentIcon
+            color={activeIndex === index ? iconColor : undefined}
+            a11yTitle={`Slide ${index + 1}`}
```
Comment (ericsoderberghp) [inline src/js/components/Carousel/Carousel.js:117]: I think we should be putting the `a11yTitle` on the icon instead of on the Button

Comment (taysea) [inline src/js/components/Carousel/Carousel.js:117]: Would it be more descriptive to say "Go to Carousel Slide 1"?

Comment (g4rry420) [reply src/js/components/Carousel/Carousel.js:117]: If we put the `a11yTitle` on the button then the icon will have `Subtract Button` a11yTitle by default.

Comment (halocline) [reply src/js/components/Carousel/Carousel.js:138]: I like either "Go to" or "Show" as they more specifically capture the action.

```suggestion
            a11yTitle={`Show carousel slide ${index + 1}`}
```

Comment (halocline) [reply src/js/components/Carousel/Carousel.js:138]: Agree with @ericsoderberghp that the `a11yTitle` should be at the Button level. 

In my testing, if the `a11yTitle` is placed on the Button, the text is read correctly as "Slide 1, button" and the "subtract, button" is no longer read.

**Review by halocline — COMMENTED** [review]
Looking super close. Minor adjustments and this should be ready to go.

File: src/js/components/Carousel/Carousel.js (hunk 2 above)
Comment (g4rry420) [reply src/js/components/Carousel/Carousel.js]: @halocline Okay, I will change it. Thank you for your help guys.

**Review by halocline — APPROVED** [review]
Thank you @g4rry420! Looks great. Appreciate your contributions!

**Review by taysea — APPROVED** [review]
Looks good, thank you!

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**